src/pages/Trade/Trade.tsx

```diff
@@ -1,18 +1,106 @@
 import { Flex } from '@chakra-ui/react'
-import { memo } from 'react'
+import { memo, useCallback, useMemo, useRef } from 'react'
+import { FormProvider, useForm } from 'react-hook-form'
 import { useTranslate } from 'react-polyglot'
+import { matchPath, Route, Switch, useHistory, useLocation } from 'react-router-dom'
 
 import { Main } from '@/components/Layout/Main'
 import { SEO } from '@/components/Layout/Seo'
+import { LimitOrder } from '@/components/MultiHopTrade/components/LimitOrder/LimitOrder'
+import { LimitOrderRoutePaths } from '@/components/MultiHopTrade/components/LimitOrder/types'
+import { Claim } from '@/components/MultiHopTrade/components/TradeInput/components/Claim/Claim'
+import { ClaimRoutePaths } from '@/components/MultiHopTrade/components/TradeInput/components/Claim/types'
 import { MultiHopTrade } from '@/components/MultiHopTrade/MultiHopTrade'
+import { TradeInputTab, TradeRoutePaths } from '@/components/MultiHopTrade/types'
+import { TRADE_ROUTE_ASSET_SPECIFIC } from '@/Routes/RoutesCommon'
 
 const padding = { base: 0, md: 8 }
 
+type MatchParams = {
```

Comment:
This type already exists in `src/components/MultiHopTrade/MultiHopTrade.tsx` - did we want to import it from there, or move it somewhere common?

Reply:
Nice catch ser, probably the latter since this is now not swapper specific but trade routes (trade, limit, claim) specific: 8c0733e8a0